go/vt/key/key.go

```diff
@@ -296,3 +305,52 @@ func ParseShardingSpec(spec string) ([]*topodatapb.KeyRange, error) {
 	}
 	return ranges, nil
 }
+
+// ParseDestination parses the string representation of a Destionation
+// of the form keyspace:shard@tablet_type. You can use a / instead of a :.
+func ParseDestination(targetString string, defaultTabletType topodatapb.TabletType) (DestinationTarget, error) {
```

Comment:
I feel like this function belongs inside executor's `parseDestinationTarget`, and we should have vcursor call that instead. It will keep this file simple. Open to counter-arguments.

Reply:
@sougou  - I think this is in the only comment I haven't addressed. I will think a bit more about it.

Reply:
I thought about this. I think it makes sense. I wanted to keep the one here as a more generic version, but now I'm thinking that the slight difference in implementation could bite us in the future. 

What do you think if I move this method to the executor and make it private. The reason for this is that I would like to have something hook into it easily in the context of the tests in `plan_test.go` that does not require a real executor. 

Then in vcursor use `ParseDestinationTarget`. 

Thoughts?

Reply:
Sounds good.